File: batches_csharp/batch_01_exact_copy/compare.cs
using System;

class Factorial {
    static int Compute(int n) {
        if (n < 0) return -1;
        int result = 1;
        for (int i = 2; i <= n; i++) result *= i;
        return result;
    }

    static void Main() {
        Console.WriteLine(Compute(5));
        Console.WriteLine(Compute(0));
    }
}

File: batches_c/batch_01_exact_copy/compare.c
#include <stdio.h>

int factorial(int n) {
    if (n < 0) return -1;
    int result = 1;
    for (int i = 2; i <= n; i++) {
        result *= i;
    }
    return result;
}

int main() {
    printf("%d\n", factorial(5));
    printf("%d\n", factorial(0));
    return 0;
}
